src/app/controllers/authentication.server.controller.js

```diff
@@ -100,9 +109,25 @@ exports.verifyPermission = (requiredPermission) =>
    * @param {function} next - Next middleware function
    */
   (req, res, next) => {
-    // Admins always have sufficient permission
-    let hasSufficientPermission =
+    const isFormAdmin =
       String(req.form.admin.id) === String(req.session.user._id)
+
+    // Forbidden if requiredPersmission is admin but user is not
+    if (!isFormAdmin && requiredPermission === PERMISSIONS.ADMIN) {
+      logger.error(
+        `User ${req.session.uer} not authorized to perform ${requiredPermission} operation on Form: ${req.form.title}.`,
+      )
+      return res.status(HttpStatus.FORBIDDEN).send({
```

Comment:
Can we neaten this up so that `return res.status(HttpStatus.FORBIDDEN)` is not repeated?

Reply:
I think its okay to return earlier if requested permission is admin but user does not have admin permission. You do not want to go through the entire function when that condition could be caught early.